src/openrct2/object/RideObject.cpp

```diff
@@ -532,3 +520,605 @@ uint8 RideObject::CalculateNumHorizontalFrames(const rct_ride_entry_vehicle * ve
+        }
+    }
+    else
+    {
+        ReadJsonVehicleInfo(context, properties);
+
+        auto swingMode = ObjectJsonHelpers::GetInteger(properties, "swingMode");
+        if (swingMode == 1)
+        {
+            _legacyType.flags |= RIDE_ENTRY_FLAG_ALTERNATIVE_SWING_MODE_1;
+        }
+        else if (swingMode == 2)
+        {
+            _legacyType.flags |= RIDE_ENTRY_FLAG_ALTERNATIVE_SWING_MODE_1;
+            _legacyType.flags |= RIDE_ENTRY_FLAG_ALTERNATIVE_SWING_MODE_2;
+        }
+
+        auto rotationMode = ObjectJsonHelpers::GetInteger(properties, "rotationMode");
+        if (rotationMode == 1)
+        {
+            _legacyType.flags |= RIDE_ENTRY_FLAG_ALTERNATIVE_ROTATION_MODE_1;
+        }
+        else if (rotationMode == 2)
+        {
+            _legacyType.flags |= RIDE_ENTRY_FLAG_ALTERNATIVE_ROTATION_MODE_2;
+        }
+
+        auto ratingMultiplier = json_object_get(properties, "ratingMultipler");
+        if (ratingMultiplier != nullptr)
+        {
+            _legacyType.excitement_multiplier = ObjectJsonHelpers::GetInteger(ratingMultiplier, "excitement");
+            _legacyType.intensity_multiplier = ObjectJsonHelpers::GetInteger(ratingMultiplier, "intensity");
+            _legacyType.nausea_multiplier = ObjectJsonHelpers::GetInteger(ratingMultiplier, "nausea");
+        }
+
+        auto availableTrackPieces = ObjectJsonHelpers::GetJsonStringArray(json_object_get(properties, "availableTrackPieces"));
+        _presetColours = ReadJsonCarColours(json_object_get(properties, "carColours"));
+
+
+        // Get loading positions
+        auto loadingPositions = json_object_get(properties, "loadingPositions");
+        auto numLoadingPositions = std::min<size_t>(json_array_size(loadingPositions), 4);
+        for (size_t i = 0; i < numLoadingPositions; i++)
+        {
+            auto positions = json_array_get(loadingPositions, i);
+            auto numPositions = json_array_size(positions);
+            if (numPositions > 0 && numPositions <= std::numeric_limits<uint16>::max())
+            {
+                std::vector<sint8> positionData;
+                if (numPositions < 255)
+                {
+                    positionData.push_back(static_cast<sint8>(numPositions));
+                }
+                else
+                {
+                    positionData.push_back(-1);
+                    positionData.push_back(static_cast<sint8>(numPositions & 0xFF));
+                    positionData.push_back(static_cast<sint8>(numPositions >> 8));
+                }
+
+                for (size_t j = 0; j < numPositions; j++)
+                {
+                    auto pos = json_integer_value(json_array_get(positions, j));
+                    pos = Math::Clamp<sint64>(std::numeric_limits<sint8>::min(), pos, std::numeric_limits<sint8>::max());
+                    positionData.push_back(pos);
+                }
+                _peepLoadingPositions[i] = std::move(positionData);
```

Comment:
This does not look correct the position data should have stripped out the number of positions.

Reply:
I think this is the issue, it was actually trying to generate and prepend the count from the json which did not have it.